src/OpenTK.Windowing.Desktop/Monitors.cs

```diff
@@ -22,29 +22,35 @@ namespace OpenTK.Windowing.Desktop
     /// </summary>
     public static class Monitors
     {
-        private static List<MonitorInfo> _monitorInfos = new List<MonitorInfo>();
+        static unsafe Monitors()
+        {
+            GLFWProvider.EnsureInitialized();
 
-        // I am not sure how well of an idea this is.
-        private static Dictionary<IntPtr, int> _monitorIndexLookup = new Dictionary<IntPtr, int>();
+            if (!GLFWProvider.IsOnMainThread)
+            {
+                throw new InvalidOperationException("Only GLFW main thread can build the monitor cache.");
+            }
 
-        private static bool _isCacheBuilt = false;
-        private static bool _isHookSet = false;
+            _monitorCallback = DpiMonitorCallback;
+            GLFW.SetMonitorCallback(_monitorCallback);
+        }
+
+        private static readonly GLFWCallbacks.MonitorCallback _monitorCallback;
+
+        /// <summary>
+        /// Occurs when a <see cref="MonitorHandle"/> is connected or disconnected.
+        /// </summary>
+        public static event Action<MonitorEventArgs> OnMonitorConnected;
 
         /// <summary>
         /// The number of monitors available.
         /// </summary>
-        public static int Count
+        [Obsolete("Use GetMonitors() instead.", true)]
+        public static unsafe int Count
         {
             get
             {
-                if (CheckCache())
-                {
-                    return _monitorInfos.Count;
-                }
-                else
-                {
-                    throw new GLFWException("This information is not cached and cannot be queried outside of the GLFW main thread");
-                }
+                return -1;
```

Comment:
I don't know how the dotnet runtime handles dynamic linking, but is there a case where this code would actually run? (since it can't build due to `IsError=true`) Shouldn't this be replaced with an exception rather than silently returning an unexpected value?

Reply:
Maybe we should put this in it's own minor version and just remove them outright with no `[Obsolete]` attribute